File: src/main/com/jason/s7/KtBase134.java
package com.jason.s7;

public class KtBase134 {
    public static void main(String[] args) {
        System.out.println(MyObject.target);

        /**
         * 添加@JvmStatic Java端可以直接調用函數，不用先調用Companion
         */
        MyObject.Companion.showLocation("Juan");
        MyObject.showLocation("Juan");
    }
}

File: src/main/com/jason/s7/KtBase134.kt
package com.jason.s7

//註解@JvmStatic與Kotlin關係
fun main() {
    MyObject.showLocation("Jason")
}

class MyObject {
    companion object {
        @JvmField
        val target = "地點1"

        @JvmStatic
        fun showLocation(name: String) = println("${name}要去${target}玩")
    }
}